### PR TITLE
TOOLS-3309 Use `aarch64` for releases on RHEL platforms

### PR DESCRIPTION
**Pull request by tdq45gj**
Updated code comment in `platform.go` and changed arch names for releases on RHEL platforms.

### CODE REVIEW

**Review by FGasper — CHANGES_REQUESTED** [review]
Can we maybe add a unit test to catch this kind of mistake in the future? Something that detects its own platform and ensures that platform.go is correct for itself?

File: release/platform/platform_test.go [hunk 1]
```diff
@@ -116,3 +116,18 @@ func TestPlatformsHaveNoDuplicates(t *testing.T) {
 		assert.Equal(t, 1, dupes, "platform %v only occurs once in platforms list", p1)
 	}
 }
+
+func TestPlatformCorrectArch(t *testing.T) {
```
Comment (FGasper) [inline release/platform/platform_test.go:120]: I was thinking something that would actually determine the local Linux distro and check that the arch is correct. But this probably suffices.

**Review by FGasper — APPROVED** [review]
LGTM

**Review by autarch — APPROVED** [review]
LGTM!

**Pull request merged**